### ISSUE TITLE
[BUG] [katana] tx status is ACCEPTED_ON_L2 even though tx failed

### ISSUE THREAD

**Issue by ClementWalter** [dojoengine/dojo#864]
**Describe the bug**

When running some tx that should fail in kakarot, I realized that even though the logs says that the tx failed, the status in the receipt is still ACCEPTED_ON_L2

**To Reproduce**

Run any tx that fails, then `get_transaction_receipt`. See example logs

```
[2023-09-05T14:05:05Z INFO  katana_core::backend::executor] Transaction received | Hash: 0x78068d5d10c8faf17112190841efd71d7db5ca8cf107b32741f57776afe591e
[2023-09-05T14:05:05Z WARN  katana_core::backend::executor] Transaction execution error: "Error in the called contract (0x06f1d4fcc236ad2b25b4526e55b015e8e01c9e42dfd90a2e466d101037491197):
    Error at pc=0:7:
    Got an exception while executing a hint.
    Cairo traceback (most recent call last):
    Unknown location (pc=0:163)
    Unknown location (pc=0:149)

    Error in the called contract (0x06f1d4fcc236ad2b25b4526e55b015e8e01c9e42dfd90a2e466d101037491197):
    Error at pc=0:31:
    Got an exception while executing a hint.
    Cairo traceback (most recent call last):
    Unknown location (pc=0:4606)
    Unknown location (pc=0:4545)
    Unknown location (pc=0:4374)
    Unknown location (pc=0:376)

    Error in the called contract (0x05c945fcd6e7ab0572fd699dab15c931ee4b154f141283767924eef26a523595):
    Error at pc=0:4551:
    Error message: Kakarot: Reverted with reason: 0

    Cairo traceback (most recent call last):
    Unknown location (pc=0:19117)
    Unknown location (pc=0:19072)
    Unknown location (pc=0:18617)
    Unknown location (pc=0:18317)
    "
[2023-09-05T14:05:05Z TRACE katana_core::backend::executor] Transaction resource usage: Steps: 725929 | Bitwise: 6299 | L1 Gas: 2448 | Pedersen: 205 | Range Checks: 27058
[2023-09-05T14:05:05Z TRACE katana_core::backend::executor] Event emitted keys=[0x99cd8bde557814842a3121e8ddfd433a539b8c9f14bf31ebf108d12e6196e9]
[2023-09-05T14:05:05Z INFO  katana_core::backend] ⛏️ Block 33 mined with 1 transactions
```

and 

```
$ starkli transaction-receipt 0x78068d5d10c8faf17112190841efd71d7db5ca8cf107b32741f57776afe591e --rpc http://0.0.0.0:5050
{
  "type": "INVOKE",
  "transaction_hash": "0x78068d5d10c8faf17112190841efd71d7db5ca8cf107b32741f57776afe591e",
  "actual_fee": "0x102c58092c1a800",
  "status": "ACCEPTED_ON_L2",
  "block_hash": "0x3370275bf0715ec88b91768c24eb6f54ca55648b81905d9b443e8ae3b71d918",
  "block_number": 33,
  "messages_sent": [],
  "events": [
    {
      "from_address": "0x49d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7",
      "keys": [
        "0x99cd8bde557814842a3121e8ddfd433a539b8c9f14bf31ebf108d12e6196e9"
      ],
      "data": [
        "0x6f1d4fcc236ad2b25b4526e55b015e8e01c9e42dfd90a2e466d101037491197",
        "0x69420",
        "0x102c58092c1a800",
        "0x0"
      ]
    }
  ]
}
```

**Expected behavior**

Status should be ~REJECTED~ REVERTED post 0.12.1 update

**Comment by kariy** [comment]
the issue here is that the status in the receipt should be `REVERTED` instead of `ACCEPTED_ON_L2`.

this line should be modified to reflect the actual status https://github.com/dojoengine/dojo/blob/d2fce1a0d0694b164d0923b6626fc81701cb9190/crates/katana/core/src/backend/storage/transaction.rs#L216

**Comment by kariy** [comment]
@ClementWalter merged `0.4.0` rpc spec in #816. can you confirm if this is still an issue

**Comment by kariy** [comment]
Closing this as complete
